src/arcs.rs

```diff
@@ -59,8 +59,9 @@ impl<T: Clone> Arcs<T> {
 
     fn do_grow(&self, new_len: usize, f: impl FnMut() -> T) {
         let slice = self.as_slice();
+        let slice_ref = &*slice;
```

Comment:
I don't think this is required, it does not improve the code, nor does it reduce amount of code.